encodings/sparse/src/compute/search_sorted.rs

```diff
@@ -206,249 +186,62 @@ mod tests {
-        let res = search_sorted(&array, 45, SearchSortedSide::Left).unwrap();
-        assert_eq!(res, expected);
-    }
-
-    #[rstest]
-    #[case(sparse_high_null_fill(), SearchResult::NotFound(19))]
-    #[case(sparse_high_non_null_fill(), SearchResult::NotFound(19))]
-    #[case(sparse_low(), SearchResult::NotFound(2))]
-    #[case(sparse_low_high(), SearchResult::NotFound(19))]
-    fn search_mid_patches_not_found_right(#[case] array: ArrayRef, #[case] expected: SearchResult) {
-        let res = search_sorted(&array, 45, SearchSortedSide::Right).unwrap();
-        assert_eq!(res, expected);
-    }
-
-    #[rstest]
-    #[should_panic]
-    #[case(sparse_high_null_fill(), Scalar::null_typed::<i32>(), SearchResult::Found(18))]
-    #[case(
-        sparse_high_non_null_fill(),
-        Scalar::primitive(22, Nullability::NonNullable),
-        SearchResult::Found(0)
-    )]
-    #[case(
-        sparse_low(),
-        Scalar::primitive(60, Nullability::NonNullable),
-        SearchResult::Found(3)
-    )]
-    #[case(
-        sparse_low_high(),
-        Scalar::primitive(30, Nullability::NonNullable),
-        SearchResult::Found(2)
-    )]
     #[case(
-        sparse_high_fill_in_patches(),
-        Scalar::primitive(33, Nullability::NonNullable),
+        high_fill_in_patches(),
+        33,
+        SearchSortedSide::Left,
         SearchResult::Found(0)
     )]
     #[case(
-        sparse_low_fill_in_patches(),
-        Scalar::primitive(55, Nullability::NonNullable),
+        low_fill_in_patches(),
+        55,
+        SearchSortedSide::Left,
         SearchResult::Found(2)
     )]
     #[case(
-        sparse_low_high_fill_in_patches_low(),
-        Scalar::primitive(22, Nullability::NonNullable),
+        low_high_fill_in_patches_low(),
+        22,
+        SearchSortedSide::Left,
         SearchResult::Found(1)
     )]
     #[case(
-        sparse_low_high_fill_in_patches_high(),
-        Scalar::primitive(33, Nullability::NonNullable),
-        SearchResult::Found(17)
-    )]
-    fn search_fill_left(
-        #[case] array: ArrayRef,
-        #[case] search: Scalar,
-        #[case] expected: SearchResult,
-    ) {
-        let res = search_sorted(&array, search, SearchSortedSide::Left).unwrap();
-        assert_eq!(res, expected);
-    }
-
-    #[rstest]
-    #[should_panic]
-    #[case(sparse_high_null_fill(), Scalar::null_typed::<i32>(), SearchResult::Found(18))]
-    #[case(
-        sparse_high_non_null_fill(),
-        Scalar::primitive(22, Nullability::NonNullable),
-        SearchResult::Found(17)
-    )]
-    #[case(
-        sparse_low(),
-        Scalar::primitive(60, Nullability::NonNullable),
-        SearchResult::Found(20)
-    )]
-    #[case(
-        sparse_low_high(),
-        Scalar::primitive(30, Nullability::NonNullable),
-        SearchResult::Found(17)
+        low_high_fill_in_patches_high(),
+        33,
+        SearchSortedSide::Left,
+        SearchResult::Found(2)
```

Comment:
this case was wrong @lwwmanning, the expected result here is Found(2) not Found(17)